client/lxd_images.go

```diff
@@ -484,6 +484,14 @@ func (r *ProtocolLXD) CreateImage(image api.ImagesPost, args *ImageCreateArgs) (
 		req.Header.Set("User-Agent", r.httpUserAgent)
 	}
 
+	if image.Source != nil && image.Source.Fingerprint != "" && image.Source.Secret != "" {
```

Comment:
Shouldn't be needed unless I forgot something

Reply:
We need this check as `image.Source` is `nil` sometimes. If I remove the check, our tests fail.